plugins/modules/nxos_vrf_interfaces.py

```diff
@@ -0,0 +1,553 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2024 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""
+The module file for nxos_vrf_interfaces
+"""
+
+from __future__ import absolute_import, division, print_function
+
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: nxos_vrf_interfaces
+short_description: Resource module to configure VRF interfaces.
+description: This module configures and manages the VRF configuration on interfaces on NX-OS platforms.
+version_added: 1.0.0
+author: Ruchi Pakhle (@Ruchip16)
+notes:
+  - Tested against Cisco NX-OS.
+  - This module works with connection C(network_cli).
+options:
+  config:
+    description: A list of interface VRF configurations.
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description:
+          - Full name of the interface excluding any logical unit number,
+            i.e. Ethernet1/1.
+        type: str
+        required: true
+      vrf_name:
```

Comment:
isn't vrf name is mandatory ?

Reply:
name is mandatory vrf_name isn't, ref: https://github.com/ansible-collections/cisco.ios/blob/main/plugins/module_utils/network/ios/argspec/vrf_interfaces/vrf_interfaces.py#L41